src/transformers/modeling_tf_utils.py

```diff
@@ -1152,6 +1153,68 @@ def __init__(self, config, *inputs, **kwargs):
     def get_config(self):
         return self.config.to_dict()
 
+    # TODO Matt: Rebase and replace all tf.keras with keras after merging the tf_keras PR.
+    #            Do not merge this before that PR is in!
+    @functools.wraps(tf.keras.Model.fit)
+    def fit(self, *args, **kwargs):
+        # Convert HF BatchEncoding objects to dicts that Keras understands
+        if args and isinstance(args[0], BatchEncoding):
+            args = list(args)
+            args[0] = dict(args[0])
+        elif "x" in kwargs and isinstance(kwargs["x"], BatchEncoding):
+            kwargs["x"] = dict(kwargs["x"])
```

Comment:
As this logic is repeated, let's abstract it out into a function

Reply:
Done!